packages/esbuild/lib/index.js

```diff
@@ -116,15 +117,15 @@ export function esbuild(options) {
           '\n'
         messages = file.messages
       } catch (error_) {
-        const cause = /** @type {VFileMessage | Error} */ (error_)
+        const cause = /** @type (VFileMessage | Error) */ (error_)
         const message =
-          'reason' in cause
-            ? cause
-            : new VFileMessage('Cannot process MDX file with esbuild', {
-                cause,
-                ruleId: 'process-error',
-                source: '@mdx-js/esbuild'
-              })
+          new VFileMessage(
+            `Cannot process MDX file with esbuild:\n  ${error_}`, {
```

Comment:
```suggestion
            'Cannot process MDX file with esbuild', {
```

The point of cause, I believe, is to be an alternative to copying it into the reason.
I’m fine with always making a new error, a vfile message.

Reply:
Okay, I changed it to not concatenate the message at this point, and instead walk the cause chain and concatenate the text when creating the esbuild message.

In theory the cause chain is available in the esbuild message via the `detail` field, but, that's a very generic field and basically nothing seems to look at it-- certainly esbuild itself doesn't print it out when reporting errors.